idunn/places/poi.py

```diff
@@ -82,7 +78,49 @@ def get_contribute_url(self):
             return None
 
     def get_source(self):
-        return self.source
+        pass
+
+
+class OsmPOI(POI):
+    @cached_property
+    def osm_id_tuple(self):
+        poi_id = self.get_id()
+        try:
+            _prefix, osm_kind, osm_id = poi_id.rsplit(":", 2)
+            return (osm_kind, osm_id)
+        except ValueError:
+            return tuple()
+
+    def get_source_url(self):
+        try:
+            osm_kind, osm_id = self.osm_id_tuple
+            return f"https://www.openstreetmap.org/{osm_kind}/{osm_id}"
+        except ValueError:
+            return None
+
+    def get_contribute_url(self):
+        try:
+            osm_kind, osm_id = self.osm_id_tuple
+            edit_params = {osm_kind: osm_id}
+            if OSM_CONTRIBUTION_HASHTAGS:
+                edit_params["hashtags"] = OSM_CONTRIBUTION_HASHTAGS
+            return f"https://www.openstreetmap.org/edit?{urlencode(edit_params)}"
+        except ValueError:
+            return None
```

Comment:
I think your intention was to also remove their implementation from the class `OsmPoi`?